### ISSUE TITLE
New release 1.46.0 broke the databricks_instance_pool azure_attribute spot_bid_max_price idempotency

### ISSUE THREAD

**Issue by seblatre** [databricks/terraform-provider-databricks#3969]
Hello, since the try to migrate to 1.46.0 (was working with 1.45.0), I'm facing an issue with `databricks_instance_pool` which doesn't keep a stable state.
Each plan leads to new change that should be applied and therefore a new instance_pool to be re-created and propagated to our cluster policy and cluster configuration.

### Configuration
I have the following simplified configuration:
```hcl
locals {
  cluster_by_key = {
    A43_multi_node_spot = {
      node_type = "pool"
      multi_node = {
        driver = {
          flavor = "Standard_DS12_v2"
        },
        worker = {
          flavor = "Standard_DS3_v2",
          min = 0,
          max = 3
        }
      }
      has_spot = true
    }
  }
}

resource "databricks_instance_pool" "pool_worker" {
  for_each = {
    for key, cluster in local.cluster_by_key :
    key => cluster if cluster.node_type == "pool" && cluster.multi_node != null
  }

  instance_pool_name                    = "Pool_${each.key}_Worker"
  min_idle_instances                    = each.value.multi_node.worker.min
  max_capacity                          = each.value.multi_node.worker.max
  node_type_id                          = each.value.multi_node.worker.flavor
  idle_instance_autotermination_minutes = 5
  enable_elastic_disk                   = true

  azure_attributes {
    availability = each.value.has_spot ? "SPOT_AZURE" : "ON_DEMAND_AZURE"
    spot_bid_max_price = 0
  }
}
```

The current state of the resource (from `terraform state show`) is the following:

```
$ terraform state show 'module.[...].databricks_instance_pool.pool_worker["A43_multi_node_std"]'
# module.[...].databricks_instance_pool.pool_worker["A43_multi_node_std"]:
resource "databricks_instance_pool" "pool_worker" {
    enable_elastic_disk                   = true
    id                                    = "0829-093851-duel12-pool-5v9078vm"
    idle_instance_autotermination_minutes = 5
    instance_pool_id                      = "0829-093851-duel12-pool-5v9078vm"
    instance_pool_name                    = "Pool_A43_multi_node_spot_Worker"
    max_capacity                          = 3
    min_idle_instances                    = 0
    node_type_id                          = "Standard_DS3_v2"

    azure_attributes {
        availability       = "SPOT_AZURE"
        spot_bid_max_price = -1
    }
}
```

### Expected Behavior
Don't have any change in the plan while nothing is changed in the configuration

### Actual Behavior
When running the plan, I get each time: 
```
  # module.[...].databricks_instance_pool.pool_worker["A43_multi_node_spot"] must be replaced
-/+ resource "databricks_instance_pool" "pool_worker" {
      ~ id                                    = "0829-093851-duel12-pool-5v9078vm" -> (known after apply)
      ~ instance_pool_id                      = "0829-093851-duel12-pool-5v9078vm" -> (known after apply)
        # (7 unchanged attributes hidden)

      ~ azure_attributes {
          - spot_bid_max_price = -1 -> null # forces replacement
            # (1 unchanged attribute hidden)
        }
    }
```

### Steps to Reproduce
1. `terraform plan`

### Terraform and provider versions
I tried with the latest databricks version 1.51.0 but the issue is appearing at 1.46.0

```
Terraform v1.9.5
on windows_amd64
+ provider registry.terraform.io/databricks/databricks v1.46.0
```

### Is it a regression?
Yes it's a regression and worked in the 1.45.0 and below

### Debug Output
I didn't find anything relevant in output while filtering on databricks keyword.

### Important Factoids
None

### Would you like to implement a fix?
No

Note: I tried with spot_bid_max_price = -1 but not working better unfortunately

How could I overcome this issue please?

**Comment by alexott** [comment]
There are two issues here:

1. the `0` value isn't sent for `spot_bid_max_price` - this will be solved with the upgrade of this resource to the plugin framework.
2. there is no `suppress_diff` on this field.  We can add it as a tactical fix.

Cc: @mgyucht

**Comment by seblatre** [comment]
Hello @alexott, not sure to get what it means. Is there anything that I can workaround on my side or should I wait for a fix?
Thanks!

**Comment by alexott** [comment]
Please wait for a fix

**Comment by seblatre** [comment]
This fix have been included in the 1.52.0. After a test with this version, it's fixing the issue!
Thanks for your work :)
